### PR TITLE
Remove legacy FileStore (resolves #1029)

### LINKED ISSUE #1029: k thx bye Job.FileStore
remove job.FileStore and retain just CachedFileStore

### PR DESCRIPTION
**Pull request by arkal**
resolves #1029

Removed legacy Job.FileStore.
Removed all mentions of the legacy cache requirement.
Removed --useSharedCache flag.
Modified worker.py to remove all legacy FileStore operations.
Renamed Job.CachedFileStore to Job.FileStore

### CODE REVIEW

**Comment by arkal** [comment]
@hannes-ucsc this is ready!

**Comment by arkal** [comment]
@hannes-ucsc this is ready!

File: src/toil/common.py [hunk 1]
```diff
@@ -661,10 +650,10 @@ def createBatchSystem(config, jobStore=None, userScript=None):
         else:
             raise RuntimeError('Unrecognised batch system: %s' % config.batchSystem)
 
-        if not (config.disableSharedCache or batchSystemClass.supportsWorkerCleanup()):
-            raise RuntimeError('%s currently does not support shared caching.  Use the '
-                               '--disableSharedCache option when running Toil with this batch '
-                               'system.' % config.batchSystem)
+        if not batchSystemClass.supportsWorkerCleanup():
+            raise RuntimeError('%s currently does not support shared caching.  Use Toil versions'
+                               '< 3.2.0 (Legacy Toil) if you want to use this batch system.' %
```
Comment (hannes-ucsc) [inline src/toil/common.py]: `Use Toil version 3.1.6 along with the --disableSharedCache if you want to …`

The notion of what is "legacy" will change over time.

File: src/toil/job.py [hunk 2]
```diff
@@ -1076,7 +597,45 @@ def open(self,job):
...
+            :rtype: string
+            """
+            handle, tmpFile = tempfile.mkstemp(prefix="tmp",
+                                               suffix=".tmp", dir=self.localTempDir)
+            os.close(handle)
+            return os.path.abspath(tmpFile)
+
+        def getLocalTempFileName(self):
+            '''
+            Get a valid name for a new local file. Do it in a really stupid way by creating and then
```
Comment (hannes-ucsc) [inline src/toil/job.py]: Move the caveat out of the docstring to a `#` comment.

Comment (hannes-ucsc) [inline src/toil/job.py]: Also, I don't think there is anything stupid about the approach.

**Comment by hannes-ucsc** [comment]
Only two nitpicks. Otherwise ok.

File: src/toil/job.py [hunk 3]
```diff
@@ -1234,7 +807,7 @@ def readGlobalFile(self, fileStoreID, userPath=None, cache=True, mutable=None):
                     # either the file was successfully downloaded and added to cache, or something
                     # failed. To prevent code duplication, we recursively call readGlobalFile.
                     flock(lockFileHandle, LOCK_UN)
-                    return self.readGlobalFile(fileStoreID, userPath, cache)
+                    return self.readGlobalFile(fileStoreID, userPath, cache, mutable)
```
Comment (hannes-ucsc) [inline src/toil/job.py]: Optional arguments must always be called by name. I hope you don't do this all over the place!

The reason is that optional arguments should be order-insensitive. When modifying a function, I should be free to reorder the optional arguments with affecting existing call sites of that function. PyCharm doesn't warn you about this?

**Comment by hannes-ucsc** [comment]
And another one.

File: src/toil/job.py (hunk 3 above)
Comment (arkal) [inline src/toil/job.py]: it didn't. maybe i've turned off that inspection?

Comment (hannes-ucsc) [inline src/toil/job.py]: PEP-8 doesn't mention it. I think I still have a valid point.

Comment (arkal) [inline src/toil/job.py]: i agree. Made the fix

**Comment by arkal** [comment]
@hannes-ucsc i implemented the suggested changes

**Pull request merged**